### DISCUSSION TITLE
Is this a good way to use the `loader`

### DISCUSSION THREAD

**Discussion by AviAvinav** [remix-run/remix#4298] [Q&A] [answered]
I am currently trying to fetch data from my cms which is running on `localhost:1337`. I am using a `loader` for it and it works perfectly fine right now but I want to know if this is really the right way to do it (I feel something's wrong). Please help me out with this.

```ts
import { fetch } from '@remix-run/node';
import { useLoaderData } from '@remix-run/react';

export const loader = async () => {
  const products = await fetch('http://localhost:1337/api/products');

  return products.json();
};

const Home = () => {
  const { data } = useLoaderData();
  console.log(data);
  return <div></div>;
};

export default Home;

```

This is my code, please ignore the `console.log`, added that to test if it is fetching the data or not

**Comment by kiliman** [comment, marked as answer]
Yes, this is the recommended way to call an external API from your loader.

For the most part, it is best if you client only fetches data from your loaders instead of calling the external API directly.

**Reply by AviAvinav** [reply]
Thanks for the help!
